### PR TITLE
Reconstruct

### PR DESCRIPTION
**Pull request by Licini**
Might look like a whole lot of changes, but the core purpose of this PR is really just:
- Migrate all shared kwargs from `BufferObject` to the base `Object`, and explicitly define them at `viewer.add()`, so that at user level, people will be able to see all acceptable parameters through pylance.

image

Some other smaller changes are:
- Removed  ambiguous visualization parameters like `show_vertices` , `show_edges`, `color` and `size`. Following visualisation parameters now apply universally across all object types: `show_points`, `show_lines`, `show_faces`, `pointcolor`, `linecolor`, `facecolor` , `pointcolors`, `linecolors`, `facecolors`, `pointsize`, `linewidth`. 
- Use `Color` class when applicable.
- Adjusted usage for `Collection` class, see `scripts/v120_collections.py`.

To discuss:
- For object specific parameters, such as `u`, `v` for `SphereObject`, should we treat them as additional kwargs or a seperate dict?  
```python
# Current setup
viewer.add(sphere, facecolor=Color(1, 0, 0), u=10, v=10)
# or somethine like
viewer.add(sphere, facecolor=Color(1, 0, 0), properties={'u': 10, 'v': 10})
```

### CODE REVIEW

File: scripts/v120_arrows.py [hunk 1]
```diff
@@ -2,12 +2,13 @@
...
 from compas_view2.shapes import Arrow
+from compas.colors import Color
 
 viewer = App()
 
 for x in range(5):
     for y in range(5):
         arrow = Arrow([x, y, 0], [0, 0, 1], head_portion=0.2, head_width=0.07, body_width=0.02)
-        viewer.add(arrow, u=16, show_edges=False, color=(random(), random(), random()))
+        viewer.add(arrow, u=16, show_lines=False, facecolor=Color(random(), random(), random()))
```
Comment (brgcode) [inline scripts/v120_arrows.py:12]: for a random color, you can just do `Color.from_i(random())`

Comment (brgcode) [reply scripts/v120_arrows.py:12]: which should probably become `Color.random()`

File: scripts/v120_colors.py [hunk 2]
```diff
@@ -16,11 +17,11 @@
...
-facecolors = {k: [random(), random(), random()] for k in mesh.faces()}
-linecolors = {k: [random(), random(), random()] for k in mesh.edges()}
-pointcolors = {k: [random(), random(), random()] for k in mesh.vertices()}
+facecolors = {k: Color(random(), random(), random()) for k in mesh.faces()}
+linecolors = {k: Color(random(), random(), random()) for k in mesh.edges()}
+pointcolors = {k: Color(random(), random(), random()) for k in mesh.vertices()}
 
 viewer.add(mesh, name="mesh1", show_points=True, facecolors=facecolors, linecolors=linecolors, pointcolors=pointcolors)
-viewer.add(mesh2, name="mesh2", show_points=True, facecolor=[1, 0, 0], linecolor=[0, 1, 0], pointcolor=[0, 0, 1])
+viewer.add(mesh2, name="mesh2", show_points=True, facecolor=Color(1, 0, 0), linecolor=Color(0, 1, 0), pointcolor=Color(0, 0, 1))
```
Comment (brgcode) [inline scripts/v120_colors.py:25]: not a big fan of the differentiation between `xxxcolor` and `xxxcolors`. what if both are set? in my opinion, `facecolor` should accept either a single color or a dict. through a descriptor this is always converted to a dict in the background. this is how the regular artists work...

File: scripts/v120_points.py [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 for i in range(10):
     point = Point(random()*10, random()*10, random()*10)
-    viewer.add(point, color=(random(), random(), random()), size=30)
+    viewer.add(point, pointcolor=Color(random(), random(), random()), pointsize=30)
 
 cloud = Pointcloud.from_bounds(5, 5, 5, 1000)
-colors = {i: [random(), random(), random()] for i in range(1000)}
-viewer.add(cloud, colors=colors)
+colors = {i: Color(random(), random(), random()) for i in range(1000)}
+viewer.add(cloud, pointcolors=colors)
```
Comment (brgcode) [inline scripts/v120_points.py:14]: same here. should just be `pointcolor` in my opinion...

File: src/compas_view2/app/app.py [hunk 4]
```diff
@@ -224,22 +224,86 @@ def resize(self, width: int, height: int):
...
+            is_visible: bool = True,
+            show_points: bool = False,
+            show_lines: bool = True,
+            show_faces: bool = True,
+            pointcolor: Color = None,
+            linecolor: Color = None,
+            facecolor: Color = None,
+            facecolors: dict = None,
+            linecolors: dict = None,
+            pointcolors: dict = None,
```
Comment (brgcode) [inline src/compas_view2/app/app.py:239]: would remove...

File: src/compas_view2/objects/bufferobject.py [hunk 5]
```diff
@@ -15,51 +15,10 @@ class BufferObject(Object):
...
-        self.linecolor = np.array(linecolor or color or self.default_color_lines, dtype=float)
-        self.facecolor = np.array(facecolor or color or self.default_color_faces, dtype=float)
-
-        self.pointcolors = pointcolors or {}
-        self.facecolors = facecolors or {}
-        self.linecolors = linecolors or {}
-
-        self.linewidth = linewidth or 1
-        self.pointsize = pointsize or 10
+    visualisation : list, read-only
```
Comment (brgcode) [inline src/compas_view2/objects/bufferobject.py:18]: can we be more precise about the contents of the list?

File: src/compas_view2/objects/object.py [hunk 6]
```diff
@@ -33,22 +37,119 @@ def _get_object_cls(data):
 class Object(ABC):
     """Base object for compas_view2
 
+    Parameters
+        ----------
```
Comment (brgcode) [inline src/compas_view2/objects/object.py:41]: something wrong with the indentation...

File: src/compas_view2/objects/object.py [hunk 7]
```diff
@@ -33,22 +37,119 @@ def _get_object_cls(data):
...
-        Whether the object is selected.
-    translation : list
-        The translation vector of the object.
-    rotation : list
-        The Euler rotation of the object in XYZ order.
-    scale : list
-        The scale factor of the object.
-    matrix: list
-        The 4x4 transformation matrix that is composed from translation, rotation and scale.
+        is_selected : bool
```
Comment (brgcode) [inline src/compas_view2/objects/object.py:91]: same

**Review by brgcode — COMMENTED** [review]
a few small things here and there

File: scripts/v120_colors.py (hunk 2 above)
Comment (Licini) [reply scripts/v120_colors.py:25]: yep i prefer way too, gonna make this change

**Comment by brgcode** [comment]
@Licini any updates here?

**Comment by Licini** [comment]
Since this involves some API changes, so I think it's better to not yet merge today but examine it more carefully then put into next release ( in very near future tho )

**Comment by Licini** [comment]
@brgcode Just cleaned up what's left, Can we take another look on this now? 
(would need to merge this first https://github.com/compas-dev/compas/pull/1062)

**Review by brgcode — APPROVED** [review]

**Pull request merged**